src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs

```diff
@@ -175,38 +179,47 @@ public override string ReadToEnd()
         //
         public override string? ReadLine()
         {
-            if (_s == null)
+            string? s = _s;
+            if (s == null)
             {
-                throw new ObjectDisposedException(null, SR.ObjectDisposed_ReaderClosed);
+                ThrowHelper.ThrowObjectDisposedException_ReaderClosed();
             }
+            var pos = _pos;
+            if ((uint)pos >= (uint)s.Length)
+                return null;
 
-            int i = _pos;
-            while (i < _length)
+            ReadOnlySpan<char> remaining = s.AsSpan(pos);
+            int foundLineLength = remaining.IndexOfAny('\r', '\n');
+            if (foundLineLength >= 0)
             {
-                char ch = _s[i];
-                if (ch == '\r' || ch == '\n')
+                string result = foundLineLength == 0
+                    ? string.Empty
+                    : new string(remaining.Slice(0, foundLineLength));
+
+                char ch = remaining[foundLineLength];
+                pos += foundLineLength + 1;
+                if (ch == '\r')
                 {
-                    string result = _s.Substring(_pos, i - _pos);
-                    _pos = i + 1;
-                    if (ch == '\r' && _pos < _length && _s[_pos] == '\n')
+                    if ((uint)pos < s.Length && s[pos] == '\n')
                     {
-                        _pos++;
+                        ++pos;
```

Comment:
Nit: we tend to use post-increment rather than pre-increment throughout our codebase.